packages/devtools-local-toolbox/public/js/index.js

```diff
@@ -0,0 +1,102 @@
+/* global window, document, DebuggerConfig */
+
+const { bindActionCreators, combineReducers } = require("redux");
+const { Provider } = require("react-redux");
+const ReactDOM = require("react-dom");
+const React = require("react");
+
+const DevToolsUtils = require("devtools-sham/shared/DevToolsUtils");
+const AppConstants = require("devtools-sham/sham/appconstants").AppConstants;
+const { injectGlobals, debugGlobal } = require("./utils/debug");
+const { isEnabled, getValue,
+        isDevelopment, setConfig } = require("../../config/feature");
+
+setConfig(DebuggerConfig);
+
+// Set various flags before requiring app code.
+if (isEnabled("logging.client")) {
+  DevToolsUtils.dumpn.wantLogging = true;
+}
+
+const client = require("./clients");
+const { getClient, connectClients, startDebugging } = client;
+
+const Root = require("./components/Root");
+
+function initApp() {
+  const configureStore = require("./utils/create-store");
+  const reducers = require("./reducers");
+  const LandingPage = require("./components/LandingPage");
+
+  const createStore = configureStore({
+    log: getValue("logging.actions"),
+    makeThunkArgs: (args, state) => {
+      return Object.assign({}, args, { client: getClient(state) });
+    }
+  });
+
+  const store = createStore(combineReducers(reducers));
+  const actions = bindActionCreators(require("./actions"), store.dispatch);
+
+  if (isDevelopment()) {
+    AppConstants.DEBUG_JS_MODULES = true;
+    injectGlobals({ store });
+  }
+
+  return { store, actions, LandingPage };
+}
+
+function renderRoot(_React, _ReactDOM, component, _store) {
```

Comment:
Yup. I dislike this API too. I'd be happy to brainstorm better APIs that line up w/ the inspector's goals and ours